.github/workflows/docker-build-push.yml

```diff
@@ -0,0 +1,47 @@
+# SPDX-FileCopyrightText: 2024 Kaushlendra Pratap <kaushlendra-pratap.singh@siemens.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+        fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME  }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Set build version
+        run: |
+          VERSION_PATTERN='([[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+)(-?rc[[:digit:]]+)?-?([[:digit:]]*)-?[[:alnum:]]*'
+          echo VERSION=$(git describe --tags > /dev/null 2>&1 && git describe --tags | head -1 | sed -re "s/${VERSION_PATTERN}/\\1.\\3\\2/" | sed -re 's/\.$/.0/') >> $GITHUB_ENV
+
+      - name: Build and push main image
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          tags: fossology/licensedb:latest
```

Comment:
Idea: use the latest release version as `latest` tag, otherwise use commit SHA.

Reply:
Yea, I was planning to have both tags, latest and version.
Will that make more sense?

Reply:
Yes, but before that, we have to start releasing.

Reply:
IMO, the `main` should build with `latest` tag and the releases with tag version as we do with FOSSology. Using commit SHA will be good if we have more users of the service.

With the current status, it will create unnecessary noise/data which no one will use.

Reply:
Fair enough. Good to go then